add generated global settings fragment

guillermooo opened the pull request:
Made substantial changes to some settings.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/guillermooo/sublime-undocs/214)

<!-- Reviewable:end -->

guillermooo committed:
generate content for global settings

show type as table

add generated content for remaining settings

changes based on review

add all settings available in Preferences

updates based on review

guillermooo commented:
- [x] default values are mostly incorrect; I need to update those
- [x] squash commits

FichteFoll commented:
Reviewed 11 of 11 files at r1.
Review status: all files reviewed at latest revision, 24 unresolved discussions.

---

_[a discussion](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdZ4fLoD2wShFi8NJJ:-KKdZ4fLoD2wShFi8NJK:-657906420) (no related file):_
All default values that are strings should be written with quotes, like so: ```"./\\()\"'-:,.;<>~!@#$%^&*|+=[]{}`~?"```.

---

_[source/_includes/settings_build.g.txt, line 51 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKd_ADJToU-oqaXlvIH:-KKd_ADJToU-oqaXlvII:159990460) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_build.g.txt#L51)):_

> ```
> 
>    +-------------------+-----------+
>    | **Type**          | bool      |
> ```

Type is a bool or a dict. In Python typing, `Union[bool, Dict[str,str]]`. That seems rather complex for beginners or those unfamiliar with the typing system :/

Or we just lie about it and say it needs to be a dict either way, since techincally the setting is just tested for its truthiness before being used.

---

_[source/_includes/settings_file.g.txt, line 97 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdZqpWmAGU1WkgdXdF:-KKdZqpWmAGU1WkgdXdG:-73700207) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_file.g.txt#L97)):_

> ```
> 
>    If ``true``,
>    Sublime Text will remove white space
> ```

"trailing white space". 

Also, "white space" or "whitespace"?

---

_[source/_includes/settings_file.g.txt, line 99 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdZxNLWYwIpyILqdXd:-KKdZxNLWYwIpyILqdXe:1766722872) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_file.g.txt#L99)):_

> ```
>    Sublime Text will remove white space
>    from the active file
>    after saving.
> ```

Before saving.

---

_[source/_includes/settings_global.g.txt, line 100 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdaa8NTvuCwDZC_Wuu:-KKdaa8NTvuCwDZC_Wuv:-1645377145) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_global.g.txt#L100)):_

> ```
>    | **Type**          | bool     |
>    +-------------------+----------+
>    | **Default value** | ``true`` |
> ```

Different default value on OSX. Like a few other settings, notably:
- scroll_past_end
- move_to_limit_on_up_down
- find_selected_text

---

_[source/_includes/settings_global.g.txt, line 121 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdb5f5jhEbcVGIxa6t:-KKdb5f64rNeBoBTKXfT:1677514811) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_global.g.txt#L121)):_

> ```
>    Files in preview
>    will be closed automatically
>    as soon as they lose focus
> ```

This description of transient tabs is incorrect.

---

_[source/_includes/settings_global.g.txt, line 188 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdaDwDIb53RCkzX6K0:-KKdaDwDIb53RCkzX6K1:-2021816821) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_global.g.txt#L188)):_

> ```
>    in a global :file:`Preferences.sublime-settings` file,
>    or in folder items
>    in a ``.sublime-project`` file.
> ```

Is this to be defined for folders? Or is this an _actual_ project setting?

---

_[source/_includes/settings_system.g.txt, line 11 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdbwl104KC375Z-ZlX:-KKdbwl104KC375Z-ZlY:-1633504093) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_system.g.txt#L11)):_

> ```
>    +-------------------+-----------+
> 
>    Returns ``true``
> ```

Should say "Holds" or "Is" instead of "Returns", since we are not describing a function but a storage.

---

_[source/_includes/settings_system.g.txt, line 36 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdcAgJMI79UN-nqwVp:-KKdcAgK_LBzwmO26OxG:-721422339) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_system.g.txt#L36)):_

> ```
> 
>    Word list to be used by the spell checker.
>    Accepts a path rooted at the data directory
> ```

"relative to the data directory"? 
`:path:`{Data}``?

---

_[source/_includes/settings_visual.g.txt, line 16 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdcw3ieRVUPcs38xo9:-KKdcw3ieRVUPcs38xoA:227618653) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L16)):_

> ```
> 
>    If ``false``,
>     only shows the rectangle
> ```

Wrong indentation for this block.

---

_[source/_includes/settings_visual.g.txt, line 31 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdd-Ccf8TR3-sm36V3:-KKdd-Ccf8TR3-sm36V4:-487520988) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L31)):_

> ```
>    Sets the colors used for text highlighting.
>    Accepts a path
>    rooted at the data directory
> ```

"rooted" again

---

_[source/_includes/settings_visual.g.txt, line 44 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdd5xU1-t64hjNdF5d:-KKdd5xU1-t64hjNdF5e:-268055166) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L44)):_

> ```
> 
>    Font face to be used
>    for editable text.
> ```

"text in views"/"buffer text"? Read-only files (?), the console or output panels are also affected by this, yet not editable.

---

_[source/_includes/settings_visual.g.txt, line 55 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKddOw8qkCq5SFpG6TN:-KKddOw8qkCq5SFpG6TO:-1582418045) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L55)):_

> ```
>    +-------------------+--------+
> 
>    Size of the font for editable text.
> ```

See above.

---

_[source/_includes/settings_visual.g.txt, line 84 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKddSeu6IMLMRU8udLB:-KKddSeu6IMLMRU8udLC:-1688220733) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L84)):_

> ```
> 
>    +-------------------+--------------+
>    | **Type**          | str | [str,] |
> ```

Types are floating point numbers. Yes, not just integers.

---

_[source/_includes/settings_visual.g.txt, line 89 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKddfO8-rgie5i1oldT:-KKddfO9n52VtuhST5sM:-1431820504) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L89)):_

> ```
>    +-------------------+--------------+
> 
>    Columns in which to display vertical rules.
> ```

"rulers"

---

_[source/_includes/settings_visual.g.txt, line 120 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKddlWU_YFDrzCbOuKd:-KKddlWU_YFDrzCbOuKe:1469454169) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L120)):_

> ```
>    +-------------------+-----------+
> 
>    If ``false``,
> ```

This setting description should be inversed.

---

_[source/_includes/settings_visual.g.txt, line 130 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKddzw25q67jSRdUHR8:-KKddzw25q67jSRdUHR9:1674161349) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L130)):_

> ```
>    | **Type**          | int     |
>    +-------------------+---------+
>    | **Default value** | ``100`` |
> ```

I think this is `0`. Same for bottom padding.

---

_[source/_includes/settings_visual.g.txt, line 169 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdeFzl_fqrpKRE1PPV:-KKdeFzmD5Le5s6TFIcd:632960277) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L169)):_

> ```
> 
> 
> ``line_numbers``
> ```

This setting should be next to the `gutter` setting.

---

_[source/_includes/settings_visual.g.txt, line 201 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdeRqqb2TAmWMvEUKz:-KKdeRqrZNaDCOs-5mUQ:-1368791662) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L201)):_

> ```
>    | **Type**          | int      |
>    +-------------------+----------+
>    | **Default value** | ``1000`` |
> ```

The default value is `0`.

---

_[source/_includes/settings_visual.g.txt, line 218 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdef_YYFnKUMuwwdwU:-KKdef_bIdjHtD63wM4P:1390454378) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L218)):_

> ```
>    | **Type**          | bool      |
>    +-------------------+-----------+
>    | **Default value** | ``false`` |
> ```

Default is `true`.

---

_[source/_includes/settings_visual.g.txt, line 236 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdekghq2QDC09w4OGo:-KKdekghq2QDC09w4OGp:1500768656) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L236)):_

> ```
> 
>    If ``true``,
>    text will be drawn centered
> ```

This is not exactly correct. The text is shifted towards the center position, but it is not "centered line by line". I think we should clarify on this, even though it is easily observed.

---

_[source/_includes/settings_visual.g.txt, line 249 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdeynv_6FIjeO2aJSw:-KKdeynv_6FIjeO2aJSx:-1544269791) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L249)):_

> ```
> 
>    If ``false``,
>    disables underlining the brackets
> ```

Since the highlighting is configurable (in the color scheme), this should read "highlighting" rather than "underlining".

---

_[source/_includes/settings_visual.g.txt, line 263 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdfcbhZZdyIPV0CsRZ:-KKdfcbkb87fDKo9DT8K:1539189582) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L263)):_

> ```
>    If ``false``,
>    brackets will only be highlighted
>    when the caret is next to one.
> ```

```
If ``true``, 
the nearest bracket pair surrounding the caret
will be highlighted.
Otherwise the caret has to be next to a bracket.
```

---

_[source/reference/settings.rst, line 25 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdbVRcXMs4EaZyNICU:-KKdbVRcXMs4EaZyNICV:-1605413908) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/reference/settings.rst#L25)):_

> ``` reStructuredText
> in :file:`Preferences.sublime-settings` and :file:`Preferences (<platform>).sublime-settings`
> (where ``<platform>`` can be any of ``Linux``, ``OSX`` or ``Windows``)
> and, where indicated, also in :file:`.sublime-project` files.
> ```

I feel this last sub-sentence is misleading. Besides `binary_file_patterns` (which I'm unsure of), the only other truly global settings mentioned need to be specified in projects per-folder, i.e. are not actually settings for projects (and not considered).

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214)_

<!-- Sent from Reviewable.io -->

guillermooo commented:
Review status: 3 of 12 files reviewed at latest revision, 24 unresolved discussions.

---

_[a discussion](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdZ4fLoD2wShFi8NJJ:-KKqUTRIy2_4urPVDKK1:146996039) (no related file):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> All default values that are strings should be written with quotes, like so: ```"./\\()\"'-:,.;<>~!@#$%^&*|+=[]{}`~?"```.
> </details>
> I disagree. I find that to be noise. Right above the value we indicate it's a string. That should be enough.

---

_[source/_includes/settings_build.g.txt, line 51 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKd_ADJToU-oqaXlvIH:-KKqUmGKJf1rkRojN2gB:-2035370488) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_build.g.txt#L51)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> Type is a bool or a dict. In Python typing, `Union[bool, Dict[str,str]]`. That seems rather complex for beginners or those unfamiliar with the typing system :/
> 
> Or we just lie about it and say it needs to be a dict either way, since techincally the setting is just tested for its truthiness before being used.
> </details>
> Used `{ str : str}` for now. Don't know what a `bool` value would be useful for?

---

_[source/_includes/settings_file.g.txt, line 97 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdZqpWmAGU1WkgdXdF:-KKqUy5tIT8EI28RCh5e:894280098) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_file.g.txt#L97)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> "trailing white space". 
> 
> Also, "white space" or "whitespace"?
> </details>
> Good question :) I'll go with whitespace.

---

_[source/_includes/settings_global.g.txt, line 100 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdaa8NTvuCwDZC_Wuu:-KKqVCqbFs8fTeeGiGB_:-557422667) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_global.g.txt#L100)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> Different default value on OSX. Like a few other settings, notably:
> - scroll_past_end
> - move_to_limit_on_up_down
> - find_selected_text
>   </details>
>   Now hopefully generated from actual setting files.

---

_[source/_includes/settings_global.g.txt, line 121 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdb5f5jhEbcVGIxa6t:-KKqVMa4dx2nhqgVNNY3:1799382157) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_global.g.txt#L121)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> This description of transient tabs is incorrect.
> </details>
> I'll add a todo.

---

_[source/_includes/settings_global.g.txt, line 188 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdaDwDIb53RCkzX6K0:-KKqVXCw5Tb10IK-4Qzu:-1280842800) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_global.g.txt#L188)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> Is this to be defined for folders? Or is this an _actual_ project setting?
> </details>
> We need to review this.

---

_[source/_includes/settings_visual.g.txt, line 44 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdd5xU1-t64hjNdF5d:-KKqWOqLds-yLzuXlf9m:-188978651) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L44)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> "text in views"/"buffer text"? Read-only files (?), the console or output panels are also affected by this, yet not editable.
> </details>
> Yeah, don't know where this comes from. I'll add a TODO.

---

_[source/_includes/settings_visual.g.txt, line 84 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKddSeu6IMLMRU8udLB:-KKqWcAVWhkVZ3WkbGod:374065506) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L84)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> Types are floating point numbers. Yes, not just integers.
> </details>
> Floats don't seem to have any effect, so I'll say `int` only.

---

_[source/_includes/settings_visual.g.txt, line 236 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdekghq2QDC09w4OGo:-KKqXmO_BrkVHKF5fGjq:1799382157) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L236)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> This is not exactly correct. The text is shifted towards the center position, but it is not "centered line by line". I think we should clarify on this, even though it is easily observed.
> </details>
> I'll add a todo.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214)_

<!-- Sent from Reviewable.io -->

FichteFoll commented:
Reviewed 3 of 9 files at r2, 7 of 7 files at r3.
Review status: all files reviewed at latest revision, 24 unresolved discussions.

---

_[a discussion](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdZ4fLoD2wShFi8NJJ:-KLIqVvokRtNoqiB7Nec:-282546408) (no related file):_
<details><summary>

<i>Previously, guillermooo (Guillermo López-Anglada) wrote…</i></summary>



> I disagree. I find that to be noise. Right above the value we indicate it's a string. That should be enough.
> </details>
> It creates ambiguity. Denoting something as `false` could then be either the boolean false or a string `"false"`. Referring to the type is not enough here imo and being explicit about subtleties like these usually pays off.

Furthermore, if you copy the exact value it would be _actually_ equal to the default value while you need to wrap quotes around it otherwise.

I feel pretty strongly about this.

---

_[a discussion](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ99LFPX4oHQtl4s-t:-KLJ99LFPX4oHQtl4s-u:1573067779) (no related file):_
Where did you add your todos?

Maybe include them on the trello card?  https://trello.com/c/156hqoeD/13-revise-settings-section

---

_[source/_includes/settings_build.g.txt, line 51 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKd_ADJToU-oqaXlvIH:-KLIsWAvyr75xlzaVyd6:-1895664851) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_build.g.txt#L51)):_
<details><summary>

<i>Previously, guillermooo (Guillermo López-Anglada) wrote…</i></summary>



> Used `{ str : str}` for now. Don't know what a `bool` value would be useful for?
> </details>
> I just mention `bool` because the default value is `false`, although no user should care about this. The only type that actually _makes sense_ for this setting is a dict and we could lie about the default value by saying the default value is `{}` (which is not supposed to be confused with `"{}"` (you might think I'm onto something)).

---

_[source/_includes/settings_build.g.txt, line 8 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLIsMT1Zn48iZL6C_i6:-KLIsMT1Zn48iZL6C_i7:-1803377082) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_build.g.txt#L8)):_

> ```
>    | **Type**    | str     |
>    +-------------+---------+
>    | **Default** | ``n/a`` |
> ```

If we denoted strings with quoation marks, this could be an empty string (`""`). Same for the following

---

_[source/_includes/settings_file.g.txt, line 211 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLItdafaazquAuclmw7:-KLItdafaazquAuclmw8:1239254023) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_file.g.txt#L211)):_

> ```
>    +-------------+------------------------------------------------+
> 
>    Scopes where auto complete
> ```

It's a "selector".

We don't really have a doc page for this yet, but it should be cross-referenced eventually.

---

_[source/_includes/settings_file.g.txt, line 266 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLIu8CL0K7Mmvse0Whm:-KLIu8CMmufajTQNXKdp:553454743) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_file.g.txt#L266)):_

> ```
> 
> 
> ``auto_complete_with_fields``
> ```

This and the following setting names are not right.

---

_[source/_includes/settings_filesystem.g.txt, line 8 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ1Q0jnW-rtOcJAtSh:-KLJ1Q0k5sC-b7h_eLFz:-1982463331) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_filesystem.g.txt#L8)):_

> ```
>    | **Type**    | str     |
>    +-------------+---------+
>    | **Default** | ``n/a`` |
> ```

See earlier regarding `n/a`.

---

_[source/_includes/settings_filesystem.g.txt, line 31 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ1WxIIEeH8Rir90yy:-KLJ1WxIIEeH8Rir90yz:56433104) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_filesystem.g.txt#L31)):_
I think we should add the motiviation behind this, too.
Something like

> Atomic save attempts to reduce data loss by only removing the orginal file when the new file was successfully written.
> Due to the file then being renamed and replace instead of being "updated", many file listeners and virtual file systems will get confused by this however, which is why it is disabled by default.

---

_[source/_includes/settings_global.g.txt, line 136 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ2Vp8PDevPS1Hh6F0:-KLJ2Vp8PDevPS1Hh6F1:-321285584) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_global.g.txt#L136)):_

> ```
>    | **Type**    | [str,]                             |
>    +-------------+------------------------------------+
>    | **Default** | ``['.svn', '.git', '.hg', 'CVS']`` |
> ```

Why change this to single quotes? JSON only allows double quotes. Same for the lists below.

---

_[source/_includes/settings_visual.g.txt, line 84 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKddSeu6IMLMRU8udLB:-KLJ4bcg1u8gU3A9lRdF:1901575365) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L84)):_
<details><summary>

<i>Previously, guillermooo (Guillermo López-Anglada) wrote…</i></summary>



> Floats don't seem to have any effect, so I'll say `int` only.
> </details>
> ![img](https://a.desu.sh/mbzgkp.png)

notice the lines in the first column

---

_[source/_includes/settings_visual.g.txt, line 183 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ4sfwdgW48UY_WFrR:-KLJ4sfxxJUKZHWdI7Da:31779302) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_visual.g.txt#L183)):_

> ```
> 
>    If ``true``,
>    highlighting lines
> ```

"highlights"

---

_[source/_includes/settings_visual.g.txt, line 257 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ56QBRi8uAO3mP69t:-KLJ56QCzlv0HqzwZ6e7:-2026286221) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_visual.g.txt#L257)):_

> ```
>    +-------------+-------+
> 
>    Increases the size of the caret.
> ```

Say "width" here?

---

_[source/_includes/settings_visual.g.txt, line 474 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ6-6mq7nt3nNqyk5L:-KLJ6-6nHX4KYkemS2Fw:-551051674) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_visual.g.txt#L474)):_

> ```
> 
> 
> ``show_definitions``
> ```

Should we add a `.. versionadded::` rule for this setting?

---

_[source/_includes/settings_visual.g.txt, line 607 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ6itOsD5M343Dsgo7:-KLJ6itPn1cMj3cUrnyz:337316177) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_visual.g.txt#L607)):_

> ```
> 
>    If ``true``,
>    tab characters will scroll left and right
> ```

"tabs", not tab characters.

---

_[source/_includes/settings_visual.g.txt, line 665 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ7-rfgV0XalAln3Wi:-KLJ7-rfgV0XalAln3Wj:997670274) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_visual.g.txt#L665)):_

> ```
>    If ``true``,
>    always prompt before reloading a file,
>    even if the file hasn't unsaved changes.
> ```

"didn't have" (or scratch the last line)

---

_[source/_includes/settings_visual.g.txt, line 669 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ7Av07UtitcEPM2FQ:-KLJ7Av1PduYLa87n4ek:675616200) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_visual.g.txt#L669)):_

> ```
>    By default,
>    prompting will only occur
>    is a file has unsaved changes.
> ```

"if"

---

_[source/_includes/settings_visual.g.txt, line 683 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ7DQj7StX5wkMYuC0:-KLJ7DQj7StX5wkMYuC1:675608760) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_visual.g.txt#L683)):_

> ```
> 
>    If ``true``,
>    and empty window will be created at startup.
> ```

"an"

---

_[source/_includes/settings_visual.g.txt, line 735 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ7Sjh0k6RHpO4adFG:-KLJ7Sjh0k6RHpO4adFH:358934621) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_visual.g.txt#L735)):_

> ```
> 
>    If ``0``,
>    Sublime Text will guess
> ```

It doesn't "guess", it just uses the number of cores.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214)_

<!-- Sent from Reviewable.io -->

guillermooo commented:
Review status: all files reviewed at latest revision, 24 unresolved discussions.

---

_[a discussion](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdZ4fLoD2wShFi8NJJ:-KLSlxGX-Bws3yx-Ebcx:523887069) (no related file):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> It creates ambiguity. Denoting something as `false` could then be either the boolean false or a string `"false"`. Referring to the type is not enough here imo and being explicit about subtleties like these usually pays off.
> 
> Furthermore, if you copy the exact value it would be _actually_ equal to the default value while you need to wrap quotes around it otherwise.
> 
> I feel pretty strongly about this.
> </details>
> I guess you have a point. However, this requires some changes to the generator for which I don't have time now, so I'll note it down for later.

---

_[source/_includes/settings_build.g.txt, line 51 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKd_ADJToU-oqaXlvIH:-KLSxRuRL4cq2gPNhmLO:859131805) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_build.g.txt#L51)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> I just mention `bool` because the default value is `false`, although no user should care about this. The only type that actually _makes sense_ for this setting is a dict and we could lie about the default value by saying the default value is `{}` (which is not supposed to be confused with `"{}"` (you might think I'm onto something)).
> </details>
> Changed to "false", but this will probably require further explanation.

---

_[source/_includes/settings_file.g.txt, line 211 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLItdafaazquAuclmw7:-KLSxojjtwLzkNXk7yBM:-499146009) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_file.g.txt#L211)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> It's a "selector".
> 
> We don't really have a doc page for this yet, but it should be cross-referenced eventually.
> </details>
> Done.

---

_[source/_includes/settings_file.g.txt, line 266 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLIu8CL0K7Mmvse0Whm:-KLSyBbU3GN_l55TKi9R:1974897973) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_file.g.txt#L266)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> This and the following setting names are not right.
> </details>
> They exist in Preferences.

---

_[source/_includes/settings_filesystem.g.txt, line 31 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ1WxIIEeH8Rir90yy:-KLSzJXwxNt3kOBzGCwE:-201446294) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_filesystem.g.txt#L31)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> I think we should add the motiviation behind this, too.
> Something like
> 
> > Atomic save attempts to reduce data loss by only removing the orginal file when the new file was successfully written.
> > Due to the file then being renamed and replace instead of being "updated", many file listeners and virtual file systems will get confused by this however, which is why it is disabled by default.
> > </details>
> > I've changed this slightly because it didn't make sense to me after looking at the default description,  but take a look again.

---

_[source/_includes/settings_global.g.txt, line 136 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ2Vp8PDevPS1Hh6F0:-KLSzr9RW_b3L-57cSwD:-1423027847) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_global.g.txt#L136)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> Why change this to single quotes? JSON only allows double quotes. Same for the lists below.
> </details>
> Because this seems to be what Python does automatically. The generator needs to get smarter. Added a todo.

---

_[source/_includes/settings_visual.g.txt, line 474 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ6-6mq7nt3nNqyk5L:-KLT-er8IQ7AM62E7ibu:2117322103) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_visual.g.txt#L474)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> Should we add a `.. versionadded::` rule for this setting?
> </details>
> Added a todo for the generator.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214)_

<!-- Sent from Reviewable.io -->

FichteFoll commented:
Reviewed 4 of 4 files at r4.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

_[a discussion](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdZ4fLoD2wShFi8NJJ:-KLTMpwO7z2lKrEUL-ce:111516750) (no related file):_
<details><summary>

<i>Previously, guillermooo (Guillermo López-Anglada) wrote…</i></summary>



> I guess you have a point. However, this requires some changes to the generator for which I don't have time now, so I'll note it down for later.
> </details>
> I have some free time atm so I might finally take a look at the generator stuff.

---

_[a discussion](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLJ99LFPX4oHQtl4s-t:-KLTMvWxCvNffuPkQcRo:1456488249) (no related file):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> Where did you add your todos?
> 
> Maybe include them on the trello card?  https://trello.com/c/156hqoeD/13-revise-settings-section
> </details>
> Added all remaining todos (that weren't already on it) to the card referenced above.

---

_[source/_includes/settings_file.g.txt, line 266 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLIu8CL0K7Mmvse0Whm:-KLTLX4Qw9ZhIfyvNaiV:-1711538982) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_file.g.txt#L266)):_
<details><summary>

<i>Previously, guillermooo (Guillermo López-Anglada) wrote…</i></summary>



> They exist in Preferences.
> </details>
> What I mean is that `auto_complete_with_fields` exists three times.

---

_[source/_includes/settings_visual.g.txt, line 84 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKddSeu6IMLMRU8udLB:-KLTM4ycvB7jbgKvGWvB:-789355012) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L84)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> ![img](https://a.desu.sh/mbzgkp.png)
> 
> notice the lines in the first column
> </details>
> All ints are also floats, so just writing `float` would be fine and less confusin for users.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214)_

<!-- Sent from Reviewable.io -->

FichteFoll commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

_[a discussion](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKdZ4fLoD2wShFi8NJJ:-KLThd6qWHxhihxIlAX3:950210238) (no related file):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> I have some free time atm so I might finally take a look at the generator stuff.
> </details>
> I pushed a branch `dev/fichte` to the generator repo because I didn't to push onto master directly. Please checkout, review & merge.

It fixes this and a couple other comments I had.

---

_[source/_includes/settings_build.g.txt, line 8 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLIsMT1Zn48iZL6C_i6:-KLThPPeN5Mb5Loa-h2G:796628090) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_build.g.txt#L8)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> If we denoted strings with quoation marks, this could be an empty string (`""`). Same for the following
> </details>
> I changed this to ```null``` in my dev branch.

---

_[source/_includes/settings_file.g.txt, line 266 [r3]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KLIu8CL0K7Mmvse0Whm:-KLTgznGGN3E8yBIBhh9:-1206996681) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/abce890cb2724c1a4e3cab62c54886ba4d9453b0/source/_includes/settings_file.g.txt#L266)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> What I mean is that `auto_complete_with_fields` exists three times.
> </details>
> Fixed in my dev branch.

---

_[source/_includes/settings_visual.g.txt, line 84 [r1]](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214#-KKddSeu6IMLMRU8udLB:-KLTh_5x9s5_ztP-Suw7:-1206996681) ([raw file](https://github.com/guillermooo/sublime-undocs/blob/5056d2880331ad98a30fcb2aac6d5ad67805e36f/source/_includes/settings_visual.g.txt#L84)):_
<details><summary>

<i>Previously, FichteFoll wrote…</i></summary>



> All ints are also floats, so just writing `float` would be fine and less confusin for users.
> </details>
> Fixed in my dev branch.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/guillermooo/sublime-undocs/214)_

<!-- Sent from Reviewable.io -->

guillermooo commented:
I think this is done. I'll rebase and merge. Thanks!

guillermooo commented:
_squash_
